x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/servicenow.tsx

```diff
@@ -6,105 +6,128 @@
  */
 
 import { lazy } from 'react';
-import { i18n } from '@kbn/i18n';
 import {
-  GenericValidationResult,
   ActionTypeModel,
   ConnectorValidationResult,
+  GenericValidationResult,
 } from '../../../../types';
 import {
   ServiceNowActionConnector,
   ServiceNowConfig,
-  ServiceNowSecrets,
+  ServiceNowITOMActionParams,
   ServiceNowITSMActionParams,
+  ServiceNowSecrets,
   ServiceNowSIRActionParams,
-  ServiceNowITOMActionParams,
 } from './types';
 import { isValidUrl } from '../../../lib/value_validators';
 import { getConnectorDescriptiveTitle, getSelectedConnectorIcon } from './helpers';
+import {
+  SERVICENOW_ITOM_DESC,
+  SERVICENOW_ITOM_TITLE,
+  SERVICENOW_ITSM_DESC,
+  SERVICENOW_ITSM_TITLE,
+  SERVICENOW_SIR_DESC,
+  SERVICENOW_SIR_TITLE,
+} from './translations';
 
 const validateConnector = async (
   action: ServiceNowActionConnector
-): Promise<ConnectorValidationResult<ServiceNowConfig, ServiceNowSecrets>> => {
+): Promise<ConnectorValidationResult<Omit<ServiceNowConfig, 'isOAuth'>, ServiceNowSecrets>> => {
   const translations = await import('./translations');
-  const configErrors = {
+
+  const configErrorsCommon = {
     apiUrl: new Array<string>(),
     usesTableApi: new Array<string>(),
   };
-  const secretsErrors = {
+
+  const configErrorsOAuth = {
+    clientId: new Array<string>(),
+    userIdentifierValue: new Array<string>(),
+    jwtKeyId: new Array<string>(),
+  };
+
+  const secretsErrorsBasicAuth = {
```

Comment:
These (configErrorsOAuth and secretsErrorsBasicAuth) could've been a single object but wanted to show the difference between them and make it ready for a refactoring.